.github/workflows/node-test-build.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js Build and Test CI
+
+on:
+  pull_request:
+    branches: 
+      - 'master'
+      - 'production-version-*'
+      - 'dev-version-*'
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Source Code
+      uses: actions/checkout@v2
+
+    - name: Use Node.js "12.x"
+      uses: actions/setup-node@v1
+      with:
+        node-version: "12.x"
+
+    - name: Install Dependencies
+      run: npm install
```

Comment:
please cache the npm dependencies to save CI time. Not too sure the correct way to do this but can try this 👇 

possible solution: https://stackoverflow.com/questions/55110729/how-do-i-cache-steps-in-github-actions